extensions/sql-database-projects/src/controllers/mainController.ts

```diff
@@ -22,14 +31,36 @@ export default class MainController implements vscode.Disposable {
 	public deactivate(): void {
 	}
 
-	public activate(): Promise<boolean> {
-		this.initializeDatabaseProjects();
-		return Promise.resolve(true);
+	public async activate(): Promise<void> {
+		await this.initializeDatabaseProjects();
+	}
+
+	private async initializeDatabaseProjects(): Promise<void> {
+		// init commands
+		vscode.commands.registerCommand('sqlDatabaseProjects.new', () => { console.log('"New Database Project" called.'); });
+		vscode.commands.registerCommand('sqlDatabaseProjects.open', async () => { this.openProjectFolder(); });
+
+		// init view
+		this.dbProjectTreeViewProvider = new SqlDatabaseProjectTreeViewProvider();
+
+		this.extensionContext.subscriptions.push(vscode.window.registerTreeDataProvider(SQL_DATABASE_PROJECTS_VIEW_ID, this.dbProjectTreeViewProvider));
 	}
 
-	private initializeDatabaseProjects(): void {
-		vscode.commands.registerCommand('sqlDatabaseProjects.new', () => { console.log('new database project called'); });
-		vscode.commands.registerCommand('sqlDatabaseProjects.open', () => { console.log('open database project called'); });
+	public async openProjectFolder(): Promise<void> {
```

Comment:
what about when user opens a folder which has sqlproject? we should probably load the tree in that case too?

Reply:
you activate the extension in this event:workspaceContains:**/*.sqlproj so we should also load the tree 

---
In reply to: [360134139](https://github.com/microsoft/azuredatastudio/pull/8639#discussion_r360134139) [](ancestors = 360134139)